### PR TITLE
Convert item combo db to use libconfig

### PR DESCRIPTION
**Pull request by Asheraf**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2529)

File: db/pre-re/item_combo_db.conf [hunk 1]
```diff
@@ -0,0 +1,1280 @@
...
+		bonus bDef, 5;
+		bonus bMdef, 5;
+		if (BaseClass == Job_Swordman)
+			bonus bDef, 6;
+	">
+},
+{
+	Items: ["Valkyrjas_Shield", "Odins_Blessing", "Fricca_Circlet"]
+	Script: <"
+		bonus bDef, 2 - getequiprefinerycnt(EQI_HAND_L) - getequiprefinerycnt(EQI_HEAD_TOP);
```
Comment (Emistry) [inline db/pre-re/item_combo_db.conf]:
```suggestion
		bonus bDef, 2 - getequiprefinerycnt(EQI_HAND_L, EQI_HEAD_TOP);
```

File: db/re/item_combo_db.conf [hunk 2]
```diff
@@ -0,0 +1,3224 @@
...
+	Script: <"
+		bonus3 bHPDrainRateRace, 11, 40, 2 + (getequiprefinerycnt(EQI_SHADOW_ARMOR, EQI_SHADOW_SHIELD, EQI_SHADOW_SHOES) / 5);
+	">
+},
+{
+	Items: ["S_Gemstone_Armor", "S_Gemstone_Shoes", "S_Gemstone_Shield", "S_Gemstone_Weapon", "S_Gemstone_Earring", "S_Gemstone_Pendent"]
+	Script: <"
+		bonus bAllStats, 1;
+		.@r = getequiprefinerycnt(EQI_SHADOW_ARMOR, EQI_SHADOW_WEAPON, EQI_SHADOW_SHIELD, EQI_SHADOW_SHOES, EQI_SHADOW_ACC_R, EQI_SHADOW_ACC_L);
+		if (.@r >= 45) {
```
Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

File: db/re/item_combo_db.conf [hunk 3]
```diff
@@ -0,0 +1,3224 @@
...
+	Script: <"
+		bonus bNoCastCancel, 1;
+		bonus bVariableCastrate, 40 - (getequiprefinerycnt(EQI_SHADOW_WEAPON, EQI_SHADOW_ACC_R, EQI_SHADOW_ACC_L));
+	">
+},
+{
+	Items: ["S_Spell_Flow_Shoes", "S_Spell_Flow_Armor", "S_Spell_Flow_Shield"]
+	Script: <"
+		bonus bNoCastCancel, 1;
+		bonus bUseSPrate, 40 - getequiprefinerycnt(EQI_SHADOW_SHIELD) - getequiprefinerycnt(EQI_SHADOW_ARMOR) - getequiprefinerycnt(EQI_SHADOW_SHOES);
```
Comment (Emistry) [inline db/re/item_combo_db.conf]:
```suggestion
		bonus bUseSPrate, 40 - getequiprefinerycnt(EQI_SHADOW_SHIELD, EQI_SHADOW_ARMOR, EQI_SHADOW_SHOES);
```

File: db/re/item_combo_db.conf [hunk 4]
```diff
@@ -0,0 +1,3224 @@
...
+			bonus bIgnoreDefRace, RC_Insect;
+	">
+},
+{
+	Items: ["S_Fishing_Weapon", "S_Penetration_Earring", "S_Penetration_Pendent"]
+	Script: <"
+		bonus bBaseAtk, 5;
+		if (getequiprefinerycnt(EQI_SHADOW_WEAPON, EQI_SHADOW_ACC_R, EQI_SHADOW_ACC_L) >= 25)
+			bonus bIgnoreDefRace, RC_Fish;
+	"
```
Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

Comment (Emistry) [inline db/re/item_combo_db.conf]: `{  }` not needed

File: db/pre-re/item_combo_db.conf [hunk 5]
```diff
@@ -0,0 +1,1280 @@
...
+	Script: <"
+		bonus2 bAddEff, Eff_Blind, 500;
+		autobonus "{ bonus bFlee,20; }", 200, 10000, BF_WEAPON, "{ specialeffect(EF_INCAGILITY, AREA, playerattached()); }";
+	">
+},
+{
+	Items: ["Valkyrjas_Shield_C", "Odins_Blessing", "Fricca_Circlet"]
+	Script: <"
+		bonus bDef, 2 - getrefine() - getequiprefinerycnt(EQI_HEAD_TOP);
+		bonus bMdef, 5 + getrefine() + getequiprefinerycnt(EQI_HEAD_TOP);
```
Comment (Emistry) [inline db/pre-re/item_combo_db.conf]:
```suggestion
		bonus bMdef, 5 + getequiprefinerycnt(EQI_HAND_L, EQI_HEAD_TOP);
```

File: db/pre-re/item_combo_db.conf [hunk 6]
```diff
@@ -0,0 +1,1280 @@
...
+	">
+},
+{
+	Items: ["Mr_Smile", "Wit_Pumpkin_Hat"]
+	Script: <"
+		bonus2 bAddMonsterDropItem, 12192, 1;
```
Comment (Emistry) [inline db/pre-re/item_combo_db.conf]:
```suggestion
		bonus2 bAddMonsterDropItem, Pumpkin_Pie, 1;
```

**Comment by Asheraf** [comment]
@Emistry 
> Btw it no longer support ItemID ? only constant string?
> 
yes, id's can be easily mistaken and they make no sense to the human reader

> anyway, might consider to standard all the constants usage?
> some are using `"constant_name"` while other `constant_name`.
> Maybe just remove all the `" "` quote.
> 
libconfig does not support constant type for values, if it's a key then quotes can be dropped else no

> avoid `getrefine()` since `getequiprefinerycnt(EQI_**)` give better readability due to multiple items combo together, its easier to read `getequiprefinerycnt()` compare to `getrefine()`

I would not add any further formatting changes beside what the jsbeautifier gives and the few syntax errors fixes, perhaps in another pr

**Comment by Emistry** [comment]
> libconfig does not support constant type for values, if it's a key then quotes can be dropped else no

sry, i meant these 
https://github.com/HerculesWS/Hercules/blob/548aa1263b69c2c5551d2c5f3d501ebc82d1e067/db/pre-re/item_combo_db.conf#L52
https://github.com/HerculesWS/Hercules/blob/548aa1263b69c2c5551d2c5f3d501ebc82d1e067/db/re/item_combo_db.conf#L130
there are more to find in the `pre-re` version, they should be standardize

**Pull request merged**